span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java

```diff
@@ -32,9 +32,10 @@ public void init(ProcessorContext context) {
       RawSpan rawSpan = converter.convert(value);
       if (null != rawSpan) {
         String traceId = HexUtils.getHex(rawSpan.getTraceId());
-        // we use the trace_id as the key so that raw_span_grouper
+        String customerId = rawSpan.getCustomerId();
+        // we use the (customer_id|trace_id) as the key so that raw_span_grouper
         // job can do a groupByKey without having to create a repartition topic
-        return new KeyValue<>(traceId, rawSpan);
+        return new KeyValue<>(String.join("|", customerId, traceId), rawSpan);
```

Comment:
Make "|" a const. Also this seems quite flimsy. Why can't we just create a Key class with the customer id and traceId as members?

Reply:
Hmm we will need a way to serialize and deserialize a custom object. Let me see if I can do that without having to define that in avro

Reply:
Aah. I see.

Reply:
Using TraceIdentity